cocos/2d/CCScene.cpp

```diff
@@ -211,17 +240,62 @@ void Scene::setPhysics3DDebugCamera(Camera* camera)
 }
 #endif
 
-#if (CC_USE_PHYSICS || (CC_USE_3D_PHYSICS && CC_ENABLE_BULLET_INTEGRATION))
+#if CC_USE_NAVMESH
+void Scene::setNavMeshDebugCamera(Camera *camera)
+{
+    CC_SAFE_RETAIN(camera);
+    CC_SAFE_RELEASE(_navMeshDebugCamera);
+    _navMeshDebugCamera = camera;
+}
+
+void Scene::addChildToNavMesh(Node* child)
+{
+    if (_navMesh)
+    {
+        std::function<void(Node*)> addToNavMeshFunc = nullptr;
+        addToNavMeshFunc = [this, &addToNavMeshFunc](Node* node) -> void
+        {
+            auto agCom = static_cast<NavMeshAgent*>(node->getComponent(NavMeshAgent::getNavMeshAgentComponentName()));
+            if (agCom)
+            {
+                agCom->onEnter();
+            }
+
+            auto obCom = static_cast<NavMeshObstacle*>(node->getComponent(NavMeshObstacle::getNavMeshObstacleComponentName()));
+            if (obCom)
+            {
+                obCom->onEnter();
+            }
+
+            auto& children = node->getChildren();
+            for (const auto &n : children) {
+                addToNavMeshFunc(n);
+            }
+        };
+
+        addToNavMeshFunc(child);
+    }
+}
+
+#endif
+
+#if (CC_USE_PHYSICS || (CC_USE_3D_PHYSICS && CC_ENABLE_BULLET_INTEGRATION) || CC_USE_NAVMESH)
 void Scene::addChild(Node* child, int zOrder, int tag)
 {
     Node::addChild(child, zOrder, tag);
     addChildToPhysicsWorld(child);
+#if CC_USE_NAVMESH
```

Comment:
The component of `NavmeshAgent` and `NavMeshObstalces` would not be added to `NavMesh` if we do it like this:
```cpp
scene = getScene();
auto node = node::create(...);
scene->addChild(node);
auto sprite = Sprite::create(...);
sprite->addComponent(NavComponent...);
node->addChild(sprite);
```